lib/fog/azurerm/models/network/virtual_network.rb

```diff
@@ -8,37 +8,70 @@ class VirtualNetwork < Fog::Model
         attribute :id
         attribute :location
         attribute :resource_group
-        attribute :dns_list
-        attribute :subnet_address_list
-        attribute :network_address_list
+        attribute :dns_servers
+        attribute :subnets
+        attribute :address_prefixes
 
         def self.parse(vnet)
           hash = {}
           hash['id'] = vnet['id']
           hash['name'] = vnet['name']
           hash['resource_group'] = vnet['id'].split('/')[4]
```

Comment:
implement get_resource_group_from_id() as a utility method. This is being used in many parser method. Reuse the code